kpops/pipeline_generator/pipeline.py

```diff
@@ -276,3 +357,32 @@ def set_pipeline_name_env_vars(base_dir: Path, path: Path) -> None:
         os.environ["pipeline_name"] = pipeline_name
         for level, parent in enumerate(path_without_file):
             os.environ[f"pipeline_name_{level}"] = parent
+
+
+# TODO: Does it belong here?
+def generate_substitution(
+    input: dict,
+    prefix: str | None = None,
+    existing_substitution: dict | None = None,
+) -> dict:
+    """Generate a complete substitution dict from a BaseModel
+
+    Finds all attributes that belong to a model and expands them to create
+    a dict containing each variable name and value to substitute with.
+
+    :param input: Dict from which to generate the substitution
+    :type input: dict
+    :param prefix: Prefix the preceeds all substitution variables, defaults to None
+    :type prefix: str, optional
+    :param substitution: existing substitution to include
+    :type substitution: dict
+    :returns: Substitution dict of all variables related to the model.
+    :rtype: dict
+    """
+    # Check if given existing substitution is a dict
+    if existing_substitution and not isinstance(existing_substitution, dict):
+        raise TypeError("Argument existing_substitution must be a dict")
+    elif not existing_substitution:
+        existing_substitution = {}
```

Comment:
For example, here you can do something like that :
```python
def __is_substitution_dict(existing_substitution: dict) -> dict
    if existing_substitution and not isinstance(existing_substitution, dict):
        raise TypeError("Argument existing_substitution must be a dict")
    elif not existing_substitution:
        return {}
```
P.S. haven't checked the syntax properly but you get the point

Reply:
done